e2e/evm/test.go

```diff
@@ -213,10 +222,57 @@ func (s *IntegrationTestSuite) TestGenericDeposit() {
 	}
 	s.Nil(err)
 
-	time.Sleep(120 * time.Second)
-
+	WaitForProposalExecuted(s, s.bridgeAddr2)
 	// Asset hash sent is stored in centrifuge asset store contract
 	exists, err := assetStoreContract2.IsCentrifugeAssetStored(hash)
 	s.Nil(err)
 	s.Equal(true, exists)
 }
+
+func WaitForProposalExecuted(s *IntegrationTestSuite, bridge common.Address) {
```

Comment:
Probably can be moved to some utility file and accept client.
Don't think IntegrationTestSuite param is needed.